### ISSUE TITLE
Accessibility Audit

### ISSUE THREAD

**Issue by StevenDufresne** [WordPress/wporg-developer#179]
Use devtools to scan all pages.

Test with keyboard.

Test with VoiceOver and NVDA.

Seek community feedback in #accessibility

**Comment by adamwoodnz** [comment]
We addressed a lot of page layout issues in [Fix accessibility issues with page layouts](https://github.com/WordPress/wporg-developer/pull/376) and [Fix WP-CLI Commands layouts accessibility](https://github.com/WordPress/wporg-developer/pull/402).

**Comment by adamwoodnz** [comment]
There are some contrast ratio issues on code reference pages with WCAG 2.1 AA

![Screenshot 2023-11-23 at 4 05 23 PM](https://github.com/WordPress/wporg-developer/assets/1017872/93357eb4-9df3-4e21-ab56-d7fdac2ab970)

**Comment by adamwoodnz** [comment]
PHP is not a valid `lang` value, used on source code blocks

![Screenshot 2023-11-23 at 4 08 10 PM](https://github.com/WordPress/wporg-developer/assets/1017872/052110e4-bcb6-4082-a6ad-c1ad86c985f7)

**Comment by adamwoodnz** [comment]
GitHub links don't meet contrast ratios with WCAG 2.1 AA

![Screenshot 2023-11-23 at 4 10 20 PM](https://github.com/WordPress/wporg-developer/assets/1017872/39c768db-2f4a-4da2-bb59-c3e299f2a9ad)

**Comment by adamwoodnz** [comment]
Duplicate link in related with no accessible text

![Screenshot 2023-11-23 at 4 14 02 PM](https://github.com/WordPress/wporg-developer/assets/1017872/4539f501-2fea-407c-bf21-9252cc5dd96c)

**Comment by adamwoodnz** [comment]
User notes form needs labels

![Screenshot 2023-11-23 at 4 15 03 PM](https://github.com/WordPress/wporg-developer/assets/1017872/6612733d-82cb-4dd5-b5e1-db60e64859b6)

![Screenshot 2023-11-23 at 4 16 54 PM](https://github.com/WordPress/wporg-developer/assets/1017872/79750089-29f0-4fc2-ba78-a5a23bcc5a57)

**Comment by adamwoodnz** [comment]
Shared the call for testing https://wordpress.slack.com/archives/C02RP4X03/p1700709677752379

**Comment by adamwoodnz** [comment]
> There are some contrast ratio issues on code reference pages with WCAG 2.1 AA
> 
> ![Screenshot 2023-11-23 at 4 05 23 PM](https://private-user-images.githubusercontent.com/1017872/285090926-93357eb4-9df3-4e21-ab56-d7fdac2ab970.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE4MDMwMzIsIm5iZiI6MTcwMTgwMjczMiwicGF0aCI6Ii8xMDE3ODcyLzI4NTA5MDkyNi05MzM1N2ViNC05ZGYzLTRlMjEtYWI1Ni1kN2ZkYWMyYWI5NzAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDVUMTg1ODUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzJlZmU3YjIyMjQ1ZTlhYzkzOTFmZTBiODRjOWQyZjcxZDViOTA1YjBkZTFiMTNjOTNiYjUzOWIxYzFkZThmYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xE3Ef7h0udn3KsUjIJ8cVvAq2ErE6QDPTWxIrurtgEg)

@WordPress/meta-design thoughts on this please? Do we expect to meet WCAG 2.1 AA? Only the blue badge passes, yellow, red and green all fail.

**Comment by jasmussen** [comment]
We do want contrast to pass. Can we use [the syntax highlighting colors on this page](https://developer.wordpress.org/redesign-test/reference/hooks/ajax_term_search_results/)? Or am I missing nuance?

**Comment by adamwoodnz** [comment]
> Or am I missing nuance?

@jasmussen I think it's the red on low opacity red that is the problem. The colors are ok on white. Our options are probably to increase font size or weight, or lower the opacity of the background further.

**Comment by adamwoodnz** [comment]
> Our options are probably to increase font size or weight, or lower the opacity of the background further.

Actually I've tried increasing font weight and size and that has no effect. The background opacity must be lowered, or the color darkened, or both. I've managed to get the contrast passing by darkening just the text:

![Screenshot 2023-12-06 at 11 44 11 AM](https://github.com/WordPress/wporg-developer/assets/1017872/619460b3-41c2-4a5a-b349-ef356d274c3d)

I think the only one that's really noticeable is the yellow. What do you think @jasmussen?

**Comment by jasmussen** [comment]
Right, thank you for the clarification. Are you using SCSS? If yes, then I think it'd be fine to use the built-in `darken` function on just the color inside the chip, to darken it enough to achieve 4.5:1 contrast. Looks like the red one needs to be slightly darker, and certainly the yellow. Perhaps the green too, that hue is usually troublesome.

Let me know if you need specific CSS color values, though, I'll take a stab. Sorry for not capturing this sooner.

**Comment by adamwoodnz** [comment]
> Are you using SCSS? If yes, then I think it'd be fine to use the built-in `darken` function on just the color inside the chip, to darken it enough to achieve 4.5:1 contrast. Looks like the red one needs to be slightly darker, and certainly the yellow. Perhaps the green too, that hue is usually troublesome.

Thanks @jasmussen, yep using SCSS `darken`. Sorry I wasn't clear, the screenshot above shows the darkened colors which pass. I was just asking if that looked ok to you 🙂

**Comment by jasmussen** [comment]
Yep that's fine. I do think we could revisit these at some point in the future, to find the best syntax highlighting values across, that work in all contexts. Perhaps we can also add nuance so the "darken" is only used when the colors are overlayed on a chip. But that's a future optimization to consider.
